### ISSUE TITLE
Issue with import and dynamic import

### ISSUE THREAD

**Issue by Ayc0** [parcel-bundler/parcel#2546]
# 🐛 bug report

In one of my files I have:

```js
import { verify } from 'jsonwebtoken';

if (process.env.MOCK === 'true') {
  import('jsonwebtoken').then(({ sign }) => {
    // add mocks
  }
}
```

When I comment the dynamic import, it works. But when both import are present I've got this error: "Cannot find module 'jsonwebtoken'".

The weird part is when I uncomment it: at the beginning, it works (with hot reload), but when I reload, it stops working

![bug-parcel-720](https://user-images.githubusercontent.com/22725671/51248106-63fc2f00-198f-11e9-99ba-75c486d9e6dd.gif)

## 🎛 Configuration (.babelrc, package.json, cli command)

```js
{
  "presets": ["@babel/preset-react"]
}
```

## 🤔 Expected Behavior

The import statement should work in both cases.

## 😯 Current Behavior

The dynamic import statement fails

`Cannot find module 'jsonwebtoken.6ec9cde9.js,../node_modules/jsonwebtoken/index.js,jsonwebtoken.6ec9cde9.map,../node_modules/jsonwebtoken/index.js'`

## 💻 Code Sample

```js
import { decode } from "jsonwebtoken";
import { shouldMock } from "./env";

// lines of code that use decode

if (shouldMock) {
  import("jsonwebtoken").then(async ({ sign }) => {
    const mockUser = await import("./mock/user.json";
    sign({ user: mockUser }, "secret");
  });
}
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | ^1.11.0
| Node             | 11.6.0
| Yarn         | 1.13.0
| Operating System | macOS Mojave 10.14.2

**Comment by DeMoorJasper** [comment]
Not sure why you'd want both dynamic and regular import in the same file (of the same module), but this is a nice reproduction that we'd definitely need to look into fixing.

**Comment by Ayc0** [comment]
@DeMoorJasper because in the same file I have everything related to the JWTs
And if the env variable `MOCK` is set to `"true"`, I want to import the mocks.

And with tree shaking, `decode` is always imported but `sign` should only be imported if `MOCK === "true"`

**Comment by mischnic** [comment]
Seems similar to #2340 (Parcel deduplicating assets too eagerly)

**Comment by zanona** [comment]
Perhaps this can help to illustrate the use case as I have experienced an issue where **module A** imports **module B**, and **C** where **B** dynamically imports **asset D**, and **C** statically imports **asset D**.

This could be represented more or less like:


```ts
// A.ts
export {default as B} from 'B';
export {default as C} from 'C';

// B.ts
export default function () {
    const D = import('./assets/D');
    ...
}
// C.ts
const D = require('./assets/D');
export default function () {
   ...
}
```

However, if I then import only B, as below, the following happens:

```ts
 `import {B} from './A` // Asset D will be bundled because A is also exporting C statically importing D 
```

The only solution I found was to not export `C` from `A`, and then import `C` separately through different means.

Also, It may be important to mention that, in this case, `B` and `C` would not be simultaneously imported on the same project as `B` is a dynamic import friendly version of the component, whilst `C` is the same as `B` but meant to be used for SSR.

So from a parcel project, it's very likely only `B` would need to be imported. But then, A is distributed separately and unaware of this.
